packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -21,6 +21,9 @@ import 'dart/package_map.dart';
 import 'devfs.dart';
 import 'device.dart';
 import 'globals.dart' as globals;
+import 'ios/native_assets.dart';
```

Comment:
I'm a bit confused by how the PRs on this are organized, but I left a comment on another similar PR about the layering violation here.

Reply:
I don't believe GitHub has a solution for stacked pull requests. So the other PRs based on top of this one show the whole diff including this one. If there's a better workflow on GitHub I'd love to learn about it.

I've looked around what to do about the layering violation here.
It doesn't really make sense to push native assets logic into the device or even the resident compiler. The resident compiler doesn't know about the target. It's a property of the hot-runner that it has a second kernel compiler (if you count the one in the flutter assemble backend as first kernel compiler). So it should be responsible for embedding the native_assets.yaml in the kernel file.
I have moved the branching based on device to `src/native_assets.dart` now, so that it includes the various `src/<os>/native_assets.dart`.